client/state/sites/guided-transfer/reducer.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GUIDED_TRANSFER_STATUS_RECEIVE,
+	GUIDED_TRANSFER_STATUS_REQUEST,
+	GUIDED_TRANSFER_STATUS_REQUEST_FAILURE,
+	GUIDED_TRANSFER_STATUS_REQUEST_SUCCESS,
+} from 'state/action-types';
+import { guidedTransferStatusSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+// Stores the status of guided transfers per site
+export const status = createReducer( {}, {
+	[ GUIDED_TRANSFER_STATUS_RECEIVE ]: ( state, action ) => Object.assign( {}, state, {
+		[ action.siteId ]: action.guidedTransferStatus,
+	} )
+}, guidedTransferStatusSchema );
+
+// Tracks whether we're fetching the status of a guided transfer for a site
+export const isFetching = createReducer( {}, {
+	[ GUIDED_TRANSFER_STATUS_REQUEST ]: ( state, action ) =>
+		Object.assign( {}, state, { [ action.siteId ]: true } ),
```

Comment:
Not sure how you feel about it since it's still a stage 2 proposal, but I find the object spread syntax to be a nice convenient shorthand to `Object.assign`:

```js
{ ...state, [ action.siteId ]: true }
```

https://github.com/sebmarkbage/ecmascript-rest-spread